app/edge/devices/components/AMTDevicesDatatable/columns/RowContext.tsx

```diff
@@ -0,0 +1,43 @@
+import {
+  createContext,
+  useContext,
+  useMemo,
+  useReducer,
+  PropsWithChildren,
+} from 'react';
+import { EnvironmentId } from 'Portainer/environments/types';
+
+interface RowContextState {
+  environmentId: EnvironmentId;
+  isLoading: boolean;
+  toggleIsLoading(): void;
+}
+
+const RowContext = createContext<RowContextState | null>(null);
+
+export interface RowProviderProps {
+  environmentId: EnvironmentId;
+}
+
+export function RowProvider({
+  environmentId,
+  children,
+}: PropsWithChildren<RowProviderProps>) {
+  const [isLoading, toggleIsLoading] = useReducer((state) => !state, false);
+
+  const state = useMemo(
+    () => ({ isLoading, toggleIsLoading, environmentId }),
+    [isLoading, toggleIsLoading]
```

Comment:
should be dependent on environmentId too
```suggestion
    [isLoading, toggleIsLoading, environmentId]
```

Reply:
done